lua/starfall/sflib.lua

```diff
@@ -701,48 +675,230 @@ if SERVER then
 		return sf
 	end
 
+	local function getCodeCache(ply, filename, hash, codeSize)
+		local cacheEntry = codecache[ply][filename] or {}
+		for k,v in pairs(cacheEntry) do
+			if v.hash == hash and v.size == codeSize then
+				return v
+			end
+		end
+		return nil
+	end
+
+	local function addCodeCache(ply, filename, hash, codeSize)
+		if codecache[ply][filename] == nil then
```

Comment:
if `ply` returns nil, surely attempting to index `filename` upon this will result in an error `index a nil value`?

Reply:
I don't believe ply can be nil when you are inside net.Receive, I might be wrong but I'll can catch that case.